### DISCUSSION TITLE
UDFs and side_effects

### DISCUSSION THREAD

**Discussion by EugeniuZ** [duckdb/duckdb#8343] [Q&A] [answered]
Hi,

Is the parameter `side_effects` used for caching the results of previous invocations of a UDF? 

If yes, how big is the cache size?

Regards,
Eugeniu

**Comment by Tishj** [comment, marked as answer]
That's not what it's for no
If the function would not have side effects and all the parameters passed to it are constant, it would get constant folded.

Setting side effects to true prevents this

**Comment by StLeoX** [comment]
If an UDF is marked as `side_effects = false`, it will provide idempotence. Typically, the `random()` UDF has no side_effect.
